### PR TITLE
[RISCV] Move RISCVVType namespace to TargetParser

### PR DESCRIPTION
**Pull request by wangpc-pp**
Clang and some middle-end optimizations may need these helper
functions.

This can reduce some duplications.

### CODE REVIEW

**Comment by topperc** [comment]
Is it possible to move them to TargetParser instead?

**Comment by wangpc-pp** [comment]
> Is it possible to move them to TargetParser instead?

Yes, I think we can. But is there any specific reason?

**Comment by topperc** [comment]
> > Is it possible to move them to TargetParser instead?
> 
> Yes, I think we can. But is there any specific reason?

The Support library shouldn't contain target specific code if possible. The only reason RISCVISAInfo is in there is because TableGen uses it.

**Comment by wangpc-pp** [comment]
Ping.

File: llvm/unittests/TargetParser/CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,7 @@ add_llvm_unittest(TargetParserTests
   Host.cpp
   TargetParserTest.cpp
   TripleTest.cpp
+  RISCVTargetParserTest.cpp
```
Comment (topperc) [inline llvm/unittests/TargetParser/CMakeLists.txt:11]: alphabetize

**Review by topperc — APPROVED** [review]
LGTM

**Comment by wangpc-pp** [comment]
There are some failures: https://lab.llvm.org/buildbot/#/builders/57/builds/33206
Fix is here: https://github.com/llvm/llvm-project/pull/84109

**Pull request merged**